src/main/java/com/celarli/commons/vfs/provider/google/GCSFileObject.java

```diff
@@ -239,22 +242,39 @@ protected void doDelete() throws Exception {
 
     @Nonnull
     @Override
-    protected OutputStream doGetOutputStream(boolean bAppend) throws Exception {
+    protected OutputStream doGetOutputStream(boolean bAppend) {
 
-        getCurrentBlob();
+        getCurrentBlob(true);
         return Channels.newOutputStream(this.currentBlob.writer());
     }
 
 
-    private void getCurrentBlob() {
+    private void getCurrentBlob(boolean detectContentType) {
 
         URLFileName urlFileName = (URLFileName) this.getName();
         String path = urlFileName.getPath();
         if (path.startsWith("/")) {
             path = path.substring(1);
         }
 
-        this.currentBlob = storage.create(BlobInfo.newBuilder(urlFileName.getHostName(), path).build());
+        BlobInfo blobInfo;
+        if (detectContentType) {
+            String fileName = getName().getBaseName();
```

Comment:
Seems useless

Reply:
Operation like copy do not set proper content type to cloud storage, it shows application/octet-stream 
i.e //fileObject.copyFrom(sourceFileObject, new AllFileSelector());

Have we tried resolving content type using alternate way?